pkg/server/download.go

```diff
@@ -355,3 +361,18 @@ func (s *server) updateDownloadSubscribers(total int64, current int64, name stri
 		s.subscribers.Delete(id)
 	}
 }
+
+// If the source and destination are on different file systems this will cause
+// an invalid cross-link device error using os.Rename. Instead, open the file
+// and then write it out
+func deviceSafeRename(tempPath string, targetLocation string) error {
+	byteData, err := os.ReadFile(tempPath)
+	if err != nil {
+		log.Error(err)
+	}
```

Comment:
if ReadFile produces an error should the function early return instead of just logging and continuing to the WriteFile?